ThinBridge/Chronos
# Add support for prohibiting file upload and download

# Which issue(s) this PR fixes:

#34

# What this PR does / why we need it:

To work the file upload/download prohibition.

# How to verify the fixed issue:

> Describe the following information to help that the tester able to do it.

## The steps to verify:

1. Check "ファイルアップロードを禁止する" / "ファイルダウンロードを禁止する" in setting.
2. Try to upload/download files.

## Expected result:

* File uploading is prohibited if  "ファイルアップロードを禁止する"  is checked.
* File downloading is prohibited if "ファイルダウンロードを禁止する" is checked.

## Tests for checking regression

### ThinApp

#### File download dialog

* `B:\` ドライブのみが選択できること
* ルートドライブ以外を選択すると、ルートドライブしか選択できない旨の警告ダイアログがでること
  * 確認手順: 
    * 設定-> ファイルマネージャ設定 -> ダウンロード/アップロードアイテムの規定のパスを`C:\`にする
    * その状態でダウンロードダイアログを開くと`B:\`しか選択できないので、適当なファイルをダウンロードしようとする
* `B:\Upload` フォルダを選択すると、アップロードフォルダはダウンロードの対象にできない旨の警告ダイアログがでること

#### File upload dialog

* `B:\` 以外が表示されていないこと
* 前回選択したフォルダがない場合、`B:\Upload` がデフォルトで選択されること
* 前回選択したフォルダがある場合、そのフォルダがデフォルトで選択されること
* ルートに指定しているドライブ以外を選択すると、ルートに指定しているドライブしか選択できない旨の警告ダイアログがでること
  * 確認手順: 
    * 設定-> ファイルマネージャ設定 -> ダウンロード/アップロードアイテムの規定のパスを`C:\`にする
    * その状態でアップロードダイアログを開くと`B:\`しか選択できないので、適当なファイルをアップロードしようとする
* トレースログが有効な時、アップロードしたファイルがログに記録されること
### ThinApp

### Native

#### File download dialog

* デフォルトで「ダウンロード」フォルダが選択されること
* `B:\` ドライブ以外が選択できること
* `B:\` ドライブ以外を選択しても問題なくダウンロードできること

#### File upload dialog

* 前回選択したフォルダがない場合、「ダウンロード」フォルダが選択されること
* 前回選択したフォルダがある場合、そのフォルダがデフォルトで選択されること
* `B:\` ドライブ以外を選択しても問題なくダウンロードできること
* トレースログが有効な時、アップロードしたファイルがログに記録されること
